lib/urlwatch/filters.py

```diff
@@ -82,17 +82,17 @@ def auto_process(cls, state, data):
         return data
 
     @classmethod
-    def process(cls, filter_kind, subfilter, state, data):
+    def process(cls, filter_kind, subfilter, state, data, baseurl):
         logger.info('Applying filter %r, subfilter %r to %s', filter_kind, subfilter, state.job.get_location())
         filtercls = cls.__subclasses__.get(filter_kind, None)
         if filtercls is None:
             raise ValueError('Unknown filter kind: %s:%s' % (filter_kind, subfilter))
-        return filtercls(state.job, state).filter(data, subfilter)
+        return filtercls(state.job, state).filter(data, baseurl, subfilter)
 
     def match(self):
         return False
 
-    def filter(self, data, subfilter=None):
+    def filter(self, data, baseurl, subfilter=None):
```

Comment:
`baseurl` sounds too specific a name for the base class. While other filters might also utilize the job url, in their context it probably doesn't make sense to call it "base" url.

To make it even more generic, I wonder if it makes sense to pass the entire `job` object into the filter, and let filters decide for themselves what they want to do. I can't recall exactly what, but I had situations where it'd be useful to have `job` in filters.